core/src/main/java/hudson/console/ConsoleAnnotatorFactory.java

```diff
@@ -80,13 +80,13 @@
      * @return
      *      null if this factory is not going to participate in the annotation of this console.
      */
-    public abstract ConsoleAnnotator newInstance(T context);
+    public abstract ConsoleAnnotator<T> newInstance(T context);
 
     /**
      * For which context type does this annotator work?
      */
-    public Class type() {
-        Type type = Types.getBaseClass(getClass(), ConsoleAnnotator.class);
+    public Class<?> type() {
+        Type type = Types.getBaseClass(getClass(), ConsoleAnnotatorFactory.class);
```

Comment:
Uh, `this` is a `ConsoleAnnotatorFactory`, not a `ConsoleAnnotator`, so `type` was always null, causing this method to always return `Object.class`, causing `_for` to always call `newInstance`, resulting in the `ClassCastException`.